### PR TITLE
Tether improvements

### LINKED ISSUE #40: Tether needs improved dinamics.
we should calculate at least 3d drag. Right now the drag is the same in all directions, length-wise it should be almost none

### PR DESCRIPTION
**Pull request by Williangalvani**
- Improve ROV attachment (fix #40)
 - Implement better drag phyisics (force against lateral movement is stronger than force against longitudinal movement)

Before:
image

After:
image

### CODE REVIEW

**Comment by Williangalvani** [comment]
This seems to lower the framerate considerably (30fps after this patch on my Ryzen 2500U laptop cpu). We should consider using GDNative instead. (I hope the issue is the gdscript overhead on top of 200 (physics steps) * 100 (tether sections) per second)

File: rope/section.gd [hunk 1]
```diff
@@ -0,0 +1,11 @@
+extends RigidBody
+
+# Should be multiplied by cross surface area of the tether, but let's keep the
+# math low for now
+var DRAG = 0.3
```
Comment (patrickelectric) [inline rope/section.gd:5]: const ?

File: rope/section.gd [hunk 2]
```diff
@@ -0,0 +1,11 @@
...
+
+func _physics_process(delta):
+	var body_frame_speeds = self.transform.basis.xform_inv(self.linear_velocity)
+	var z_speed = body_frame_speeds.dot(Vector3(0,0,1))
```
Comment (patrickelectric) [inline rope/section.gd:9]: run the gdscript code format script. (spece after comma, check next line)

**Review by patrickelectric — APPROVED** [review]

**Comment by Williangalvani** [comment]
I believe I had a heavy scene in the background while tested the gdscript one.
it turned out that, at 500 links, The gain was of 2 fps, from 19 to 22 fps.

The branch is [here](https://github.com/Williangalvani/bluesim-1/tree/tethergdnative)

**Pull request merged**